crypto/txscript/src/viewer.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{multi_sig::get_multisig_params, opcodes::codes, parse_script, TxScriptError};
+use kaspa_consensus_core::{hashing::sighash::SigHashReusedValues, tx::VerifiableTransaction};
+use std::{
+    fmt::{Display, Formatter},
+    marker::PhantomData,
+};
+
+pub struct ScriptViewer<'a, T, Reused> {
+    script: &'a [u8],
+    _phantom: PhantomData<(T, Reused)>,
+}
+
+impl<'a, T, Reused> ScriptViewer<'a, T, Reused>
+where
+    T: VerifiableTransaction,
+    Reused: SigHashReusedValues,
+{
+    pub fn new(script: &'a [u8]) -> Self {
+        Self { script, _phantom: PhantomData }
+    }
+
+    pub fn try_to_string(&self) -> Result<String, TxScriptError> {
+        let opcodes: Vec<_> = parse_script::<T, Reused>(self.script).collect::<Result<_, _>>()?;
+        let mut s = String::new();
+        let mut indent_level: usize = 0;
+
+        for (i, opcode) in opcodes.iter().enumerate() {
+            let value = opcode.value();
+
+            if value == codes::OpEndIf || value == codes::OpElse {
+                indent_level = indent_level.saturating_sub(1);
+            }
+
+            s.push_str(&"  ".repeat(indent_level));
+
+            s.push_str(opcode.to_string().as_str());
+
+            if (codes::OpData1..=codes::OpData75).contains(&value) {
+                let data = opcode.get_data();
+                s.push(' ');
+                s.push_str(&hex::encode(data));
+            } else if value == codes::OpPushData1 || value == codes::OpPushData2 || value == codes::OpPushData4 {
+                let data = opcode.get_data();
+                s.push(' ');
+                s.push_str(&data.len().to_string());
+                s.push(' ');
+                s.push_str(&hex::encode(data));
+
+                // try to disassemble the data as a script
```

Comment:
maybe it needs explicit param to try disassemble. also output is not clearly specifies that redeem script and last push data are the same. maybe it makes sense to wrap it into `: (redeem script)` or something like that

Reply:
I meant formatting.currently it looks like
```
OpCode, OpCodeData data
Redeem script 
```

There's no direct link between data and redeem script. I suggest adding some extra delimiters to make it look like `OpData hex: (redeem script)`